swo_aws_extension/flows/jobs/billing_journal/generators/authorization.py

```
@@ -63,29 +70,46 @@ def run(self, authorization: dict) -> AuthorizationJournalResult:
+        self,
+        aws_client: AWSClient,
+        pma_account: str,
+        mpa_account: str,
+    ) -> BaseOrganizationUsageGenerator:
+        if self._context.usage_source == BillingJournalUsageSourceEnum.COST_USAGE_REPORT:
+            bucket = S3_BILLING_EXPORT_BUCKET_TEMPLATE.format(pm_account_id=pma_account)
+            prefix = S3_BILLING_EXPORT_PREFIX_TEMPLATE.format(mpa_account_id=mpa_account)
+            return CostUsageReportGenerator(aws_client, bucket, prefix)
+        return CostExplorerUsageGenerator(aws_client)
+
+    def _process_agreements(  # noqa: WPS210 Found too many local variables
```

Ruff (0.15.7)

[warning] Invalid rule code in `# noqa`: WPS210

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)